sdk/src/main/scala/io/horizen/account/state/AccountStateView.scala

```diff
@@ -22,12 +23,12 @@ import java.math.BigInteger
 //  - StateDbAccountStateView (concrete class) : evm stateDb read/write
 //      Inherits its methods
 class AccountStateView(
-    metadataStorageView: AccountStateMetadataStorageView,
-    stateDb: StateDB,
-    messageProcessors: Seq[MessageProcessor]
-) extends StateDbAccountStateView(stateDb, messageProcessors)
-      with StateView[SidechainTypes#SCAT]
-      with SparkzLogging {
+  metadataStorageView: AccountStateMetadataStorageView,
+  stateDb: StateDB,
+  messageProcessors: Seq[MessageProcessor],
+) extends StateDbAccountStateView(stateDb, messageProcessors, metadataStorageView)
```

Comment:
I don't like to pass the metadataStorageView to the StateDbAccountStateView, that class must only be a view of the stateDb!
Looks you have done this to access the metadatastorage from the message processors, since they just receive the BaseStateView in the process() method.
But I think we can do the following:
- update the process() interface to add also a new parameter representing a "reader" of the  metadatastorage (can be the trait [MsgProcessorMetadataStorageReader you already added)
- update the hiearchy up to [here ](https://github.com/HorizenOfficial/Sidechains-SDK/blob/1d412b7e2488488e1b585a54841e095b4163718d/sdk/src/main/scala/io/horizen/account/api/rpc/service/EthService.scala#L230) to extract the reader from the nodeView.   IMPORTANT: in this method the stateview can be also one in the past (depends on the tag), but we can't do something like that for the metadatastorage, so the new parameter will always refer to the tip. I think it should not be a problem if we access just the rewards like now, given that we always add new items for the rewards and never update the ones in the past, but something to be checked and higlighted with a comment maybe.

Reply:
I agree with Paolo. In addition, It would be better to do some tests with the rpc methods, just to be sure that there is no problem with the metadataStorage being always the latest while the stabedb is referring to a block in the past.

Reply:
Fixed, this change introduces a lot of changes because .process is used widely. I thought about adding an override with a metadata instead of changing a method itself, at least for TestContext, but this change is bad for future maintainability of the code, so probably better to just change everything now